src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -2418,6 +2418,10 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
             {
                 pePdbFilePath = pePdbFilePath ?? FileNameUtilities.ChangeExtension(SourceModule.Name, "pdb");
             }
+            else
+            {
+                pePdbFilePath = null;
+            }
 
             if (moduleBeingBuilt.DebugInformationFormat == DebugInformationFormat.Embedded && !string.IsNullOrEmpty(pePdbFilePath))
```

Comment:
>!string.IsNullOrEmpty(pePdbFilePath) [](start = 94, length = 36)

Can this condition ever be false?

Reply:
Some inspection and guessing says "maybe, but probably not" - however, the logic involved in that conclusion kind of complex, and considering the source of this bug is bad copy-pasting, I believe we should leave the check in.